src/lepton/jpgcoder.cc

```diff
@@ -1763,19 +1765,14 @@ void decode_image_stream()
 void decode_image(const std::vector<uint8_t> &inbuf, std::vector<uint8_t> &outbuf) 
 {
     auto saved_str_in = str_in;
-    auto saved_str_out = str_out;
-
     Sirikata::VectorReader input_stream(inbuf);
     Sirikata::VectorWriter output_stream(outbuf);
     bounded_iostream bounded_outstream(&output_stream, &nop, Sirikata::JpegAllocator<uint8_t>());
 
     str_in = &input_stream;
-    str_out = &bounded_outstream;
-    decode_image_stream();
+    decode_image_stream(&bounded_outstream);
     bounded_outstream.flush();
-
     str_in = saved_str_in;
-    str_out = saved_str_out;
```

Comment:
Global `str_out` should be removed now?

Reply:
Global stream variables are still used in high-level code dealing with files, so they can't be removed.

Reply:
I removed use of global stream variables only from functions called (directly or indirectly) from encode_image/decode_image